src/geo.c

```diff
@@ -143,32 +143,17 @@ double extractUnitOrReply(client *c, robj *unit) {
 }
 
 /* Input Argument Helper.
- * Extract the distance from the specified two arguments starting at 'argv'
- * that should be in the form: <number> <unit>, and return the distance in the
- * specified unit on success. *conversions is populated with the coefficient
- * to use in order to convert meters to the unit.
- *
+ * Extract the conversion from the specified argument.
  * On error a value less than zero is returned. */
-double extractDistanceOrReply(client *c, robj **argv,
+double extractConversionOrReply(client *c, robj *argv,
```

Comment:
i think i would have rather to leave the original `extractDistanceOrReply` intact instead of what you did which was copy the majority of the code out of it (and repeat it 3 times), i suppose so that you can re-use just the conversion part for the 4th place.

i.e. i vote to restore the original extractDistanceOrReply (to be used in 3 places), and add a similar extractBoxOrReply (to be used once, and clear out some of the clutter from the "bybox" portion.